Calibration/HcalAlCaRecoProducers/src/AlCaIsoTracksProducer.cc

```diff
@@ -208,354 +208,274 @@ AlCaIsoTracksProducer::AlCaIsoTracksProducer(const edm::ParameterSet& iConfig)
+						    nNearTRKs, false);
+	LogDebug("HcalIsoTrack") << "This track : " << nTracks 
+				 << " (pt|eta|phi|p) :"  << pTrack->pt() 
+				 << "|" << pTrack->eta() << "|" 
+				 << pTrack->phi() << "|" << t_p
+				 << "e_MIP " << eMipDR 
+				 << " Chg Isolation " << hmaxNearP;
+	if (t_p>pTrackMin_ && eMipDR<eEcalMax_ && hmaxNearP<eIsolation_) {
+	  reco::HcalIsolatedTrackCandidate newCandidate(v4);
+	  newCandidate.SetMaxP(hmaxNearP);
+	  newCandidate.SetEnergyEcal(eMipDR);
+	  newCandidate.setL1(ptL1,etaL1,phiL1);
+	  newCandidate.SetEtaPhiEcal((trkDetItr->pointECAL).eta(),
+				     (trkDetItr->pointECAL).phi());
+	  HcalDetId detId = HcalDetId(trkDetItr->detIdHCAL);
+	  newCandidate.SetEtaPhiHcal((trkDetItr->pointHCAL).eta(),
+				     (trkDetItr->pointHCAL).phi(),
+				     detId.ieta(), detId.iphi());
+	  int indx(0);
+	  reco::TrackCollection::const_iterator trkItr1;
+	  for (trkItr1=trkCollection->begin(); trkItr1 != trkCollection->end(); ++trkItr1,++indx) {
+	    const reco::Track* pTrack1 = &(*trkItr1);
+	    if (pTrack1 == pTrack) {
+	      reco::TrackRef tRef = reco::TrackRef(trkCollection,indx);
+	      newCandidate.setTrack(tRef);
+	      break;
 	    }
 	  }
-	
-	nisotr++;
-	
-      } //if (noNeutrals....
-  
-  } // end of main track loop
-  
-  if(outputTColl->size() > 0)
-    {
-      //   Take HO collection
-      edm::Handle<HORecHitCollection> ho;
-      iEvent.getByToken(tok_ho_,ho);
-      
-      const HORecHitCollection Hitho = *(ho.product());
-      for(HORecHitCollection::const_iterator hoItr=Hitho.begin(); hoItr!=Hitho.end(); hoItr++)
-	{
-	  outputHOColl->push_back(*hoItr);
+	  trackCollection->push_back(newCandidate);
 	}
-           
-      // Take Preshower collection      
-      
-      // get the ps geometry
-      edm::ESHandle<CaloGeometry> geoHandle;
-      iSetup.get<CaloGeometryRecord>().get(geoHandle);
-      
-      // get the ps topology
-      EcalPreshowerTopology psTopology(geoHandle);
-      
-      // process rechits
-      edm::Handle<EcalRecHitCollection> pRecHits;
-      iEvent.getByToken(tok_ps_,pRecHits);
-      const EcalRecHitCollection& psrechits = *(pRecHits.product());
+      }
+    }
+  }
+  return trackCollection;
+}
 
-      typedef EcalRecHitCollection::const_iterator IT;
-      
-      for(IT i=psrechits.begin(); i!=psrechits.end(); i++) 
-	{
-	  outputESColl->push_back( *i );
-	}
-    }  
-
-  iEvent.put( outputHcalIsoTrackColl, "HcalIsolatedTrackCollection");
-  iEvent.put( outputTColl, "IsoTrackTracksCollection");
-  iEvent.put( outputExTColl, "IsoTrackExtraTracksCollection");
-  iEvent.put( outputEColl, "IsoTrackEcalRecHitCollection");
-  iEvent.put( outputESColl, "IsoTrackPSEcalRecHitCollection");
-  iEvent.put( outputHColl, "IsoTrackHBHERecHitCollection");
-  iEvent.put( outputHOColl, "IsoTrackHORecHitCollection");
+void AlCaIsoTracksProducer::setPtEtaPhi(std::vector< edm::Ref<l1extra::L1JetParticleCollection> >& objref, double &ptL1, double &etaL1, double &phiL1) {
```

Comment:
Can the first parameter of setPtEtaPhi be declared "const"? It looks like it should be, but, at this late stage, don't change this line of code unless other code changes are also necessary.